src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,383 +15,261 @@ public static class NUnitTestExtractorApp
-                            categoryNames = new List<string>();
-                        }
-                    }
-                }
-            }
-            else
+            // if the parent directory doesn't exist, then create it
+            if (!fi.Directory.Exists)
             {
-                throw new ArgumentNullException("dlls", "no dlls were specified");
+                fi.Directory.Create();
             }
         }
 
         /// <summary>
-        /// Returns the properly formatted test string depending on the level specified. 
+        /// Get a list of namespaces/classes/functions/testcases from the assembly
         /// </summary>
-        /// <param name="level">Used in the switch statement to determine how much info to put into the data string</param>
-        /// <param name="type">Used to get the namespace of the test</param>
-        /// <param name="methodInfo">Used to get the test info such as declaring type and name</param>
-        /// <param name="testCase">Used to get arguments of a testcase</param>
-        /// <param name="data">The string which is modified and then returned</param>
-        /// <returns>Returns properly formatted test string</returns>
-        private static string FormattedTest(Level level, Type type, MethodInfo methodInfo, TestCaseAttribute testCase, string data)
+        /// <param name="assemblyFullPath"></param>
+        public static void GenerateInfoFromAssembly(string assemblyFullPath)
         {
-            switch (level)
-            {
-                case Level.Namespace:
-                    data = type.Namespace;
-                    break;
+            HashSet<string> testInfo = new HashSet<string>();
 
-                case Level.Class:
-                    data = methodInfo.DeclaringType.ToString();
-                    break;
-
-                case Level.Function:
-                    data = methodInfo.DeclaringType + "." + methodInfo.Name;
-                    break;
-
-                case Level.TestCase:
-                    if (testCase != null)
-                    {
-                        data = FormattedTestCase(methodInfo.DeclaringType.ToString(), methodInfo.Name, testCase.Arguments, data);
-                    }
+            foreach (var item in GetValidTestSuiteTypesFromAssembly(assemblyFullPath))
+            {
+                Level lv = ParseLevel(_options.Level);
 
-                    break;
+                switch (lv)
+                {
+                    case Level.Namespace:
+                        testInfo.Add(item.Namespace);
+                        break;
+                    case Level.Class:
+                        testInfo.Add(item.FullName);
+                        break;
+                    case Level.Function:
+                        testInfo.UnionWith(GetValidTestsFromTestSuite(item));
+                        break;
+                    case Level.TestCase:
+                        break;
```

Comment:
[Should] Until TestCase is working maybe you should throw a NotImplementedException?
Although TestCase was working in the original version.